### PR TITLE
Add datasets description

### PR DESCRIPTION
**Pull request by saghiles**
### Checklist:

- [ ] I have added tests.
- [ ] I have updated the documentation accordingly.
- [ ] I have updated `README.md` (if you are adding a new model).
- [ ] I have updated `examples/README.md` (if you are adding a new example).

### CODE REVIEW

**Comment by tqtg** [comment]
@saghiles FilmTrust dataset is not in the table

File: cornac/datasets/README.md [hunk 1]
```diff
@@ -0,0 +1,154 @@
+# Datasets
+
+For convenience Cornac offers easy access to a number of popular benchmark datasets. These are listed below along with their basic characteristics, followed by a usage example. In addition to preference feedback, some of these datasets come with item and/or user auxiliary information, which are grouped into three main categories:
```
Comment (tqtg) [inline cornac/datasets/README.md]: For easy experimentation, Cornac offers access to a number of popular recommendation benchmark datasets. These are listed below along with their basic characteristics, followed by a usage example. In addition to preference feedback, some of these datasets come with the item's and/or user's auxiliary information, which are grouped into three main categories:

File: cornac/datasets/README.md [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+- **Text**. Refers to texts associated with items or users. The usual format of this data is `(item_id, text)`, or for users `(user_id, text)`. Concrete examples of such information are item textual descriptions, product reviews, movie plots, and user's reviews, just to name a few.
```
Comment (tqtg) [inline cornac/datasets/README.md]: **Text** refers to textual information associated with items or users. The usual format of this data is `(item_id, text)`, or `(user_id, text)`. Concrete examples of such information are item textual descriptions, product reviews, movie plots, and user reviews, just to name a few.

File: cornac/datasets/README.md [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+- **Graph**. For items this corresponds to a network where nodes (or vertices) are items, and links (or edges) represent relations among items. This information is typically represented by an adjacency matrix in the sparse format: `(item_id, item_id, weight)`, or simply `(item_id, item_id)` in which case the weights are assumed to be binary, i.e., 1 for observed item-pairs and 0 for missing pairs. Similarly, relations between users may be available (e.g., social network).
```
Comment (tqtg) [inline cornac/datasets/README.md]: **Graph**, for items, corresponds to a network where nodes (or vertices) are items, and links (or edges) represent relations among items. This information is typically represented by an adjacency matrix, which can be presented by triplet format `(item_id, item_id, weight)`, or simply `(item_id, item_id)` in the case of an unweighted graph. Similarly, relations between users (e.g., social networks) might be available and can be represented accordingly.

File: cornac/datasets/README.md [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+- **Text**. Refers to texts associated with items or users. The usual format of this data is `(item_id, text)`, or for users `(user_id, text)`. Concrete examples of such information are item textual descriptions, product reviews, movie plots, and user's reviews, just to name a few.
+- **Graph**. For items this corresponds to a network where nodes (or vertices) are items, and links (or edges) represent relations among items. This information is typically represented by an adjacency matrix in the sparse format: `(item_id, item_id, weight)`, or simply `(item_id, item_id)` in which case the weights are assumed to be binary, i.e., 1 for observed item-pairs and 0 for missing pairs. Similarly, relations between users may be available (e.g., social network).    
+- **Image**. Consists of visual information paired with either users or items. The common format for this type of auxiliary data is `(object_id, array)`, where object stands for user or item, the array may contain pixel intensities, in the case of raw images, or some continuous visual features, e.g., learned using deep neural nets. For instance, the Amazon clothing dataset includes product CNN features.
```
Comment (tqtg) [inline cornac/datasets/README.md]: **Image** consists of visual information paired with either users or items. The common format for this type of auxiliary data is `(object_id, ndarray)`, where object_id could be user_id or item_id, the ndarray might be the raw images, or some form of visual feature vectors extracted from the images (e.g., using learned deep convolutional neural nets). For instance, the Amazon Clothing dataset contains visual features extracted from product images using deep CNN.

File: cornac/datasets/README.md [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+
+**How to cite.** If you are using one of the datasets listed below in your research, please follow the citation guidelines by the authors (the "source" link below) of each respective dataset.  
+<table>
```
Comment (tqtg) [inline cornac/datasets/README.md]: <table>
  <tr>
    <th rowspan="2"><br>Dataset</th>
    <th colspan="4">Preference Info.</th>
    <th colspan="3">Item Auxiliary Info.</th>
    <th>User Auxiliary Info.</th>
  </tr>
  <tr>
    <td>#Users</td>
    <td>#Items</td>
    <td>#Interactions</td>
    <td>Type</td>
    <td>Text</td>
    <td>Graph</td>
    <td>Image</td>
    <td align="center">Graph</td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.amazon_clothing">Amazon Clothing</a><br>(<a href="http://jmcauley.ucsd.edu/data/amazon/"><i>origin</i></a>)</td>
    <td align="right">5,377</td>
    <td align="right">3,393</td>
    <td align="right">13,689</td>
    <td align="center">INT<br>[1,5]</td>
    <td align="center">&#10004;</td>
    <td align="center">&#10004;</td>
    <td align="center">&#10004;</td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.amazon_office">Amazon Office</a><br>(<a href="http://jmcauley.ucsd.edu/data/amazon/"><i>origin</i></a>)</td>
    <td align="right">3,703</td>
    <td align="right">6,523</td>
    <td align="right">53,282</td>
    <td align="center">INT<br>[1,5]</td>
    <td></td>
    <td align="center">&#10004;</td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.amazon_toy">Amazon Toy</a><br>(<a href="http://jmcauley.ucsd.edu/data/amazon/"><i>origin</i></a>)</td>
    <td align="right">19,412</td>
    <td align="right">11,924</td>
    <td align="right">167,597</td>
    <td align="center">INT<br>[1,5]</td>
    <td></td>
    <td></td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.citeulike">Citeulike</a><br>(<a href="http://www.wanghao.in/CDL.htm"><i>origin</i></a>)</td>
    <td align="right">5,551</td>
    <td align="right">16,980</td>
    <td align="right">210,537</td>
    <td align="center">BIN<br>{0,1}</td>
    <td align="center">&#10004;</td>
    <td></td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.epinions">Epinions</a><br>(<a href="http://www.trustlet.org/downloaded_epinions.html"><i>origin</i></a>)</td>
    <td align="right">40,163</td>
    <td align="right">139,738</td>
    <td align="right">664,824</td>
    <td align="center">INT<br>[1,5]</td>
    <td></td>
    <td></td>
    <td></td>
    <td align="center">&#10004;</td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.movielens">MovieLens 100k</a><br>(<a href="https://grouplens.org/datasets/movielens/"><i>origin</i></a>)</td>
    <td align="right">943</td>
    <td align="right">1,682</td>
    <td align="right">100,000</td>
    <td align="center">INT<br>[1,5]</td>
    <td align="center">&#10004;</td>
    <td></td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.movielens">MovieLens 1M</a><br>(<a href="https://grouplens.org/datasets/movielens/"><i>origin</i></a>)</td>
    <td align="right">6,040</td>
    <td align="right">3,706</td>
    <td align="right">1,000,209</td>
    <td align="center">INT<br>[1,5]</td>
    <td align="center">&#10004;</td>
    <td></td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.netflix">Netflix Small</a><br>(<a href="https://www.kaggle.com/netflix-inc/netflix-prize-data/"><i>origin</i></a>)</td>
    <td align="right">10,000</td>
    <td align="right">5,000</td>
    <td align="right">607,803</td>
    <td align="center">INT<br>[1,5]</td>
    <td></td>
    <td></td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.netflix">Neflix Original</a><br>(<a href="https://www.kaggle.com/netflix-inc/netflix-prize-data/"><i>origin</i></a>)</td>
    <td align="right">480,189</td>
    <td align="right">17,770</td>
    <td align="right">100,480,507</td>
    <td align="center">INT<br>[1,5]</td>
    <td></td>
    <td></td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td><a href="https://cornac.readthedocs.io/en/latest/datasets.html#module-cornac.datasets.tradesy">Tradesy</a><br>(<a href="http://jmcauley.ucsd.edu/data/tradesy/"><i>origin</i></a>)</td>
    <td align="right">19,243</td>
    <td align="right">165,906</td>
    <td align="right">394,421</td>
    <td align="center">BIN<br>{0,1}</td>
    <td></td>
    <td></td>
    <td align="center">&#10004;</td>
    <td></td>
  </tr>
</table>

File: cornac/datasets/README.md [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+- **Graph**. For items this corresponds to a network where nodes (or vertices) are items, and links (or edges) represent relations among items. This information is typically represented by an adjacency matrix in the sparse format: `(item_id, item_id, weight)`, or simply `(item_id, item_id)` in which case the weights are assumed to be binary, i.e., 1 for observed item-pairs and 0 for missing pairs. Similarly, relations between users may be available (e.g., social network).    
+- **Image**. Consists of visual information paired with either users or items. The common format for this type of auxiliary data is `(object_id, array)`, where object stands for user or item, the array may contain pixel intensities, in the case of raw images, or some continuous visual features, e.g., learned using deep neural nets. For instance, the Amazon clothing dataset includes product CNN features.
+
+**How to cite.** If you are using one of the datasets listed below in your research, please follow the citation guidelines by the authors (the "source" link below) of each respective dataset.
```
Comment (tqtg) [inline cornac/datasets/README.md]: **How to cite.** If you are using one of the datasets listed below in your research, please follow the citation guidelines by the authors (the "origin" link below) of each respective dataset.

File: cornac/datasets/README.md [hunk 7]
```diff
@@ -0,0 +1,154 @@
...
+    <td></td>
+    <td></td>
+    <td align="center">&#10004;</td>
+    <td></td>
+  </tr>
+</table>
+
+## Usage example
+
+Assume that we are interested in the FilmTrust dataset, which comes with both user-item ratings and user-user trust information. We can load these two pieces of information as follows,
```
Comment (tqtg) [inline cornac/datasets/README.md]: Assuming that we are interested in the FilmTrust dataset, which comes with both `user-item ratings` and `user-user trust` information. We can load these two parts of the data as follows

File: cornac/datasets/README.md [hunk 8]
```diff
@@ -0,0 +1,154 @@
...
+```Python
+from cornac.datasets import filmtrust
+
+ratings = filmtrust.load_feedback()
+trust = filmtrust.load_trust()
+```
+
+The rantings are in the range `[0.5,4]`, and the trust network is undirected with binary weight, i.e., `1` for observed user-user pairs, and `0` for missing ones. Here are samples, the first three lines of `rantings` and `trust`, from our dataset,
```
Comment (tqtg) [inline cornac/datasets/README.md]: The ranting values are in the range of `[0.5, 4]`, and the trust network is a binary undirected graph. Below are a few samples from our dataset

File: cornac/datasets/README.md [hunk 9]
```diff
@@ -0,0 +1,154 @@
...
+```Python
+Sample of ratings: [('1', '1', 2.0), ('1', '2', 4.0), ('1', '3', 3.5)]
```
Comment (tqtg) [inline cornac/datasets/README.md]: Samples of ratings

File: cornac/datasets/README.md [hunk 10]
```diff
@@ -0,0 +1,154 @@
...
+from cornac.datasets import filmtrust
+
+ratings = filmtrust.load_feedback()
+trust = filmtrust.load_trust()
+```
+
+The rantings are in the range `[0.5,4]`, and the trust network is undirected with binary weight, i.e., `1` for observed user-user pairs, and `0` for missing ones. Here are samples, the first three lines of `rantings` and `trust`, from our dataset,
+```Python
+Sample of ratings: [('1', '1', 2.0), ('1', '2', 4.0), ('1', '3', 3.5)] 
+Sample of trust relations: [('2', '966', 1.0), ('2', '104', 1.0), ('5', '1509', 1.0)]
```
Comment (tqtg) [inline cornac/datasets/README.md]: Samples of trust relations

File: cornac/datasets/README.md [hunk 11]
```diff
@@ -0,0 +1,154 @@
...
+```
+We can now start using our dataset. The [sorec_filmtrust](../../examples/sorec_filmtrust.py) example illustrates how to perform an experiment to evaluate the [SoRec](../models/sorec/) model on FilmTrust. Useful details on how to load the different dataset are available in the [documentation](https://cornac.readthedocs.io/en/latest/datasets.html).
```
Comment (tqtg) [inline cornac/datasets/README.md]: After being loaded, the dataset is now ready to use for model training and evaluation. For a concrete example, [sorec_filmtrust](../../examples/sorec_filmtrust.py) illustrates how to perform an experiment with [SoRec](../models/sorec/) model on FilmTrust. More details on other datasets and how to use them are available in the [documentation](https://cornac.readthedocs.io/en/latest/datasets.html).

File: cornac/datasets/README.md (hunk 1 above)
Comment (saghiles) [reply cornac/datasets/README.md]: Thanks, I revised the beginning. I choose not to use `'s` in the last sentence, it is not required here.

File: cornac/datasets/README.md (hunk 3 above)
Comment (saghiles) [reply cornac/datasets/README.md]: Thanks, I rephrased this part, mainly according to your comments.

File: cornac/datasets/README.md (hunk 4 above)
Comment (saghiles) [reply cornac/datasets/README.md]: Thanks, I revised this part taking into account you comment while reducing redundancy/repetitions.

File: cornac/datasets/README.md (hunk 5 above)
Comment (saghiles) [reply cornac/datasets/README.md]: I added the parentheses. I choose to keep "source" as it is more relevant here, thanks for this suggestion though.

File: cornac/datasets/README.md (hunk 8 above)
Comment (saghiles) [reply cornac/datasets/README.md]: Thanks, I rephrased.

File: cornac/datasets/README.md (hunk 10 above)
Comment (saghiles) [reply cornac/datasets/README.md]: I changed to "Samples from trust" instead.

File: cornac/datasets/README.md (hunk 11 above)
Comment (saghiles) [reply cornac/datasets/README.md]: Thanks, I rephrased

File: cornac/datasets/README.md (hunk 7 above)
Comment (saghiles) [reply cornac/datasets/README.md]: Noted, I did minor changes, but mainly kept the original text.

File: cornac/datasets/README.md [hunk 12]
```diff
@@ -0,0 +1,164 @@
...
+```Python
+from cornac.datasets import filmtrust
+
+ratings = filmtrust.load_feedback()
+trust = filmtrust.load_trust()
+```
+
+The ranting values are in the range `[0.5,4]`, and the trust network is undirected. Here are samples from our dataset,
+```
+Sample from ratings: [('1', '1', 2.0), ('1', '2', 4.0), ('1', '3', 3.5)]
```
Comment (tqtg) [inline cornac/datasets/README.md]: ~Sample~`Samples` from ratings: [('1', '1', 2.0), ('1', '2', 4.0), ('1', '3', 3.5)] 
~Sample~`Samples` from trust: [('2', '966', 1.0), ('2', '104', 1.0), ('5', '1509', 1.0)]

**Comment by tqtg** [comment]
LGTM

**Pull request merged**